### PR TITLE
Method added to ConfigHandler to add a new tool during initialization

### PR DESCRIPTION
**Pull request by eliasyishak**
Intended to be part of https://github.com/eliasyishak/dash_analytics/issues/5

A new method has been added to the `ConfigHandler` class that will allow for new tools being used by the user to get added to the configuration file. To handle this, the tool will take the `tool` string and check if it exists in the output from the `parseConfig()` method. If it doesn't exist, it will edit the file and enter a new entry for that tool at the bottom of the configuration file with version number = 1 and today's date.

The value will be defaulted to 1, future functionality will be added to increment the tool version if the version passed in is not actually 1.

### CODE REVIEW

File: lib/src/analytics.dart [hunk 1]
```diff
@@ -27,6 +29,18 @@ class Analytics {
...
     );
     initializer.run();
+
+    // Initialize the config handler class and check if the
+    // tool message and version have been updated from what
+    // is in the current file; if there is a new message version
+    // make the necessary updates
+    // ignore: unused_local_variable TODO: remove after increment method implemented
+    bool messagePrinted = false;
+    if (!_configHandler.parsedTools.containsKey(tool)) {
```
Comment (Jasguerrero) [inline lib/src/analytics.dart:108]: Does this need a test?

Comment (eliasyishak) [reply lib/src/analytics.dart:108]: Good point, I was going to add them soon, but I will add them here first

File: lib/src/analytics.dart [hunk 2]
```diff
@@ -27,6 +29,18 @@ class Analytics {
...
+      _configHandler.addTool(tool: tool);
+      stdout.writeln(toolsMessage);
```
Comment (christopherfujino) [inline lib/src/analytics.dart]: I don't think you can send this to stdout. Consider if the client is running in machine mode, and supposed to output valid JSON to stdout.

In hixie's original doc, the analytics API exposed two public methods that clients could call: `shouldShowMessage()` and `didShowMessage()`

Comment (eliasyishak) [reply lib/src/analytics.dart]: Ah yes, good point, thanks for pointing that out, we can have the client itself decide how to show the message appropriately

**Review by Jasguerrero — APPROVED** [review]
LGTM

**Pull request merged**